engine/src/main/java/io/zeebe/engine/processor/AsyncSnapshotDirector.java

```diff
@@ -116,51 +118,51 @@ private void prepareTakingSnapshot() {
   }
 
   private void takeSnapshot() {
-    pendingSnapshot = true;
-    createSnapshot(snapshotController::takeTempSnapshot);
+    final var tempSnapshotPosition = lowerBoundSnapshotPosition;
+    final var commitPosition = logStream.getCommitPosition();
+
+    pendingSnapshot =
+        createSnapshot(() -> snapshotController.takeTempSnapshot(tempSnapshotPosition));
+    LOG.info(LOG_MSG_WAIT_UNTIL_COMMITTED, tempSnapshotPosition, commitPosition);
 
     final ActorFuture<Long> lastWrittenPosition = streamProcessor.getLastWrittenPositionAsync();
     actor.runOnCompletion(
         lastWrittenPosition,
         (endPosition, error) -> {
           if (error == null) {
-            final long commitPosition = logStream.getCommitPosition();
             lastWrittenEventPosition = endPosition;
-
-            LOG.debug(LOG_MSG_WAIT_UNTIL_COMMITTED, endPosition, commitPosition);
             onCommitCheck();
-
           } else {
-            pendingSnapshot = false;
+            pendingSnapshot = null;
             LOG.error(ERROR_MSG_ON_RESOLVE_WRITTEN_POS, error);
           }
         });
   }
 
-  private void createSnapshot(Runnable snapshotCreation) {
-    final long start = System.currentTimeMillis();
-    snapshotCreation.run();
+  private Snapshot createSnapshot(final Supplier<Snapshot> snapshotCreation) {
+    final var start = System.currentTimeMillis();
```

Comment:
Probably we do not need this anymore since we have the metrics?

Reply:
We have metrics for our own snapshots?

Reply:
No but for the atomix snapshots which we then can use ? idk

Reply:
We still take snapshots in two different places =/ Actually here might be interesting to add a metric :sweat_smile:

Reply:
Yep agree would be nice - I think the atomix snapshot metrics is not used anymore we could remove it then as well. We can create a follow up issue for that.